pystac_client/collection_search.py

```diff
@@ -110,6 +105,29 @@ def collection_matches(
             for collection_temporal_interval in collection_temporal_extent.intervals
         )
     )
+    return bbox_overlaps, datetime_overlaps
+
+
+def collection_matches(
+    collection_dict: dict[str, Any],
+    bbox: BBox | None = None,
+    temporal_interval_str: Datetime | None = None,
+    q: str | None = None,
+) -> bool:
+    # check for overlap between provided bbox and the collection's spatial extent
+    try:
+        extent = Extent.from_dict(collection_dict.get("extent", {}))
+    except Exception:
+        warnings.warn(
+            f"Unable to parse extent from collection={collection_dict.get('id', None)}",
+            PystacClientWarning,
+        )
+        bbox_overlaps = True
+        datetime_overlaps = True
```

Comment:
I don't love defaulting to `True`, but I guess a false positive match is better than a false negative (which would still be better than an `Exception`)!

Reply:
Yeah, I went back-and-forth here. I went for `True` because, for me, a "default" Collection has global extents and `[None, None]` temporal interval ... coming up with a non-matching default is hard (Null Island plus `datetime.datetime.now`?).